src/new-components/avatar/avatar.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+
+import { Avatar } from './avatar'
+
+describe('Avatar', () => {
+    it('renders a background image when avatarUrl is supplied', () => {
+        render(getAvatar({ avatarUrl: 'https://foo.bar/com.png' }))
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+    })
+
+    it('renders initials of user name when avatarUrl is not supplied', () => {
+        render(getAvatar())
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+        expect(avatar.textContent).toBe('HM')
```

Comment:
Can you try with this custom matcher and see if it works?

```suggestion
        expect(avatar).toHaveTextContent('HM')
```